### ISSUE TITLE
Feature request: Stream output

### ISSUE THREAD

**Issue by luthfielaroeha** [npiv/chatblade#12]
Would be better if the output can be streamed so the user can get immediate result from his query without waiting until the entire response ready.

**Comment by npiv** [comment]
https://user-images.githubusercontent.com/452020/226837805-6dab1e67-c6c5-4d5f-bf00-b0b8e1564c62.mov

This seems to be working pretty well. I have it on a branch, will play with it a bit and merge to main in coming days.
You can try it for yourself if you would like:
https://github.com/npiv/chatblade/tree/feature/poc-streaming

**Comment by npiv** [comment]
Nice idea, this works pretty well. It is on this branch if you would like to play with it. https://github.com/npiv/chatblade/tree/feature/poc-streaming

just check out the branch and `pip install .`

I will test a few more things and merge to main in coming days.

**Comment by luthfielaroeha** [comment]
Wow cool man, it works!

Btw, is it possible to automatically output the result as markdown?

I just playing around with the `Live` feature from python `Rich` library, and seems like we can automatically output the result to markdown. 
```python
        with Live(console=Console(), auto_refresh=False) as live:
            for message in result:
                live.update(Markdown(message.content), refresh=True)
                ...
```

But I'm not sure if this will break another thing

**Comment by npiv** [comment]
No worries, I've merged the streaming into main, and released it under 0.1.0. I might come back with some improvements in future depending on how it is received. Thanks for the suggestion.

Will close for now. I don't consider having valid markdown during streaming a high priority myself, but am happy to accept PRs if people care to look at it.
